core/src/main/java/org/elasticsearch/cluster/service/InternalClusterService.java

```diff
@@ -482,8 +482,14 @@ public void run() {
                 // we publish here before we send a notification to all the listeners, since if it fails
                 // we don't want to notify
                 if (newClusterState.nodes().localNodeMaster()) {
-                    logger.debug("publishing cluster state version {}", newClusterState.version());
-                    discoveryService.publish(clusterChangedEvent, ackListener);
+                    logger.debug("publishing cluster state version [{}]", newClusterState.version());
+                    try {
+                        discoveryService.publish(clusterChangedEvent, ackListener);
+                    } catch (Throwable t) {
```

Comment:
I guess we need that to catch the `FailedToCommitException`? If so, why not only catch that?